### DISCUSSION TITLE
Development Process Changes

### DISCUSSION THREAD

**Discussion by schrock** [jellyfin/Swiftfin#714] [Questions] [answered]
Could we have a discussion about the current development and release process for Swiftfin? I really appreciate all the work the Swiftfin dev team has put in thus far, and I really like the app (as well as the larger Jellyfin system). However, I think it might be prudent to make some changes to how the Swiftfin development team incorporates changes and releases them, especially now that Swiftfin has been officially released on the Apple Store.

Before Swiftfin was on the Apple Store, it might have made sense to make large pull requests with sweeping, unrelated changes and bug-fixes, but I don't think that method works as well now. I would much prefer smaller, more focused pull requests that address single issues or features. This would allow for more frequent releases, which could address priority bugs and features, vastly improving the end user experience in a relatively short time. In contrast, bundling bug fixes in larger pull requests with unrelated code changes means they will take much longer to be released and reach the end user.

There are two good examples of this which I recently discovered through my own use of Swiftfin:

1. The movie library is missing from the Swiftfin interface for many users on Apple TV and sometimes iPhone, documented [here](https://github.com/jellyfin/Swiftfin/issues/621). According to the issue, it could be easily fixed with a one-line code change, but the actual fix has been placed in a [gargantuan pull request](https://github.com/jellyfin/Swiftfin/pull/593) created back in September 2022 that is still in draft.
2. Files with Dolby TrueHD audio currently output no audio in Swiftfin due to a bug in the device profile builder. There is currently a pull request with a [refactored device profile builder](https://github.com/jellyfin/Swiftfin/pull/519) along with a [simpler bug-fix pull request](https://github.com/jellyfin/Swiftfin/pull/700). However, the latter was closed and the former will not be merged until the previously linked [large pull request](https://github.com/jellyfin/Swiftfin/pull/593) (still in draft) is merged first.

I should also note that while iPhone users can always use the Jellyfin browser client as an alternative to Swiftfin, no such option exists for Apple TV users because Apple does not allow browsers on that platform.

I want to make it clear that I greatly appreciate all the work the Swiftfin team is doing on this application, and I realize that all of this work is being done for free in their personal time. Having said that, I think prioritizing work on critical bug fixes and separating that work from larger features and refactoring efforts makes more sense now and will be greatly appreciated by end users of Swiftfin.

**Comment by LePips** [comment, marked as answer]
Hey all, sorry for not responding earlier. The gist of it is: life gets in the way sometimes and nothing you can do but keep moving forward.

**Comment by rakuyoMo** [comment]
I'm also thinking about this, Swiftfin hasn't released a new version on the App Store for almost a year. I compiled the source code and found that many problems had been solved but not released.

Can a new version be released?
